### ISSUE TITLE
[device-port] [sagit] Xiaomi Mi 6

### ISSUE THREAD

**Issue by archeYR** [Halium/projectmanagement#172]
Tree: halium-7.1

- [x] Create [manifest](https://github.com/Halium/halium-devices/pull/190)
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [x] LXC container starts and does not crash
- [x] libhybris tests
  * [x] test_gps
  * [x] test_hwcomposer
  * [x] test_lights
  * [x] test_vibrator
  * [x] test_wifi
  * [x] test_sensors
  * [x] test_audio
  * [x] test_camera
  * [x] test_input
  * [ ] test_nfc
  * [ ] test_recorder

All thanks to @Verevka for getting this port that far.

**Comment by Gravityhub** [comment]
Are you working to continue this project ?

**Comment by archeYR** [comment]
Currently I'm not working on it. Last time there was (another) blocker, that I couldn't figure out.

Here is how it goes when booting hybris-boot, if anyone is interested: https://paste.ubuntu.com/p/KPFVFzghZs/

**Comment by Gravityhub** [comment]
Ok thank you anyway

**Comment by degdag** [comment]
So would you work again on this project or you give up , 
If you still intersted we can works together on this , i also doing mainelining of this device and for now i can boot 5.6-rc7 on this device with working fb and display and ufs storage as well as the power button , curren processe is getting usb to work on maineline , there is also a workin linux destro (postmarketos) that use mi6 kernel and  seem to be quit good enouth to start from it if the problem is in the kernel , i alredy test the halium initrd with that kernel and they booted just fine ;)

**Comment by archeYR** [comment]
Very nice! I'm especially interested in getting mainline kernel working, do you have any repo for your work? :) And yeah, in this case I'm interested getting Halium to work sooner. It is also interesting that you could get it further with the kernel from pmOS port, how well it works?

**Comment by degdag** [comment]
Well i have problem with internet thouse days so i couldn't upstram my work but you can easly get it . My work based on oneplus 5 . Well if you want any info you can ask me i will do my best .
If you wan to try the mainline kernel just clone onepluse mainline kernel from github and i will give you the dts file for mi6

**Comment by degdag** [comment]
The kernel from pmos not stable as what we want 
usb networking : works 
Touchscreen     : works 
Display               : works
Otg                      : works 
Tested UI (working)            : mate , xfce4 , gnome , plasma desktop (not mobile)
Tested ui (not working) : the rest

**Comment by archeYR** [comment]
Yes, a dts would be nice, thanks :) As for the Halium port, there is [another one](https://github.com/UbPorts-sagit) based on the halium-9.0 tree. It already boots to UT, although many things are not working yet.

**Comment by degdag** [comment]
Well here is my repo now working
https://github.com/degdag/mi6-mainline

**Comment by degdag** [comment]
> Yes, a dts would be nice, thanks :) As for the Halium port, there is [another one](https://github.com/UbPorts-sagit) based on the halium-9.0 tree. It already boots to UT, although many things are not working yet.

So what's working and what's not yet ?
I realy interested to run UT on my mi 6

**Comment by archeYR** [comment]
What works:
RIL (Calls, SMS), Dual-SIM, touchscreen, power button, Bluetooth (unstable).
What doesn't:
WLAN, modem (internet connection), GPS, brightness control, rotation, vibration, flash light, camera, volume buttons, front buttons, and probably more...

**Comment by degdag** [comment]
i managed to get usb to work on mainline this night and yes usb is workin now .
the next is to get  usb otg and touchscreen to work

**Comment by Verevka** [comment]
> i managed to get usb to work on mainline this night and yes usb is workin now .
> the next is to get usb otg and touchscreen to work

Hi i am working on ubuntu touch on halium-7.1, 9.0.  9.0 is very good.  7.1 does not work bluetooth and wifi.  Can you use your kernel somehow?

**Comment by Verevka** [comment]
> What works:
> RIL (Calls, SMS), Dual-SIM, touchscreen, power button, Bluetooth (unstable).
> What doesn't:
> WLAN, modem (internet connection), GPS, brightness control, rotation, vibration, flash light, camera, volume buttons, front buttons, and probably more...

Almost everything works now except for usb connection and auto brightness.

**Comment by Verevka** [comment]
halium 7.1 work ril, call, mobile internet.-

https://github.com/UbPorts-sagit/android_device_xiaomi_msm8998-common/commit/5c529abd515e150bccfb56d21eb1d864ffc64d96

**Comment by Verevka** [comment]
Camera halium 7.1 fix. Work!

**Comment by Verevka** [comment]
Wifi work :-)

**Comment by Verevka** [comment]
Gps work :-)

**Comment by Verevka** [comment]
Fix torch, flashlight, video decoding, offline changer... and more fix

**Comment by degdag** [comment]
> > i managed to get usb to work on mainline this night and yes usb is workin now .
> > the next is to get usb otg and touchscreen to work
> 
> Hi i am working on ubuntu touch on halium-7.1, 9.0. 9.0 is very good. 7.1 does not work bluetooth and wifi. Can you use your kernel somehow?

Well I can give you a compiled kernel to test it if u want
